### ISSUE TITLE
Extract a regular daily/recent changelog based on https://github.com/openshiftio/saas-openshiftio

### ISSUE THREAD

**Issue by maxandersen** [openshiftio/openshift.io#1420]
With changes coming into osio from multiple places it can be hard if not impossible to track where changes come from.

There are so many that automation is necessary and idea is to use 
 https://github.com/openshiftio/saas-openshiftio as a starting point.

Basically traverse the structure in that repo, for each found repository take the sha's and fetch the changelog from that SHA until the previous SHA's and dump the commit log into one or more markdown/asciidoc files  in a repo to which we can link to by a static url.

**Comment by joshuawilson** [comment]
who is going to own this?

**Comment by maxandersen** [comment]
at the moment me - talking with KB on candidates.

**Comment by maxandersen** [comment]
@jaseemabid is looking at this.

idea is to add a script to https://github.com/openshiftio/saasherder which will generate a changelog based on the commit history  on that and connected repos.

**Comment by joshuawilson** [comment]
For the UI repos, we use Semantic Release. This adds a nice layer of organization to the changelog that is generated and viewable on the release page on github. 
If all the teams used it, then we would have cleaner commits.

**Comment by maxandersen** [comment]
> If all the teams used it, then we would have cleaner commits.

Yes - and once we get this aggregator in it will become very apparent :) 

Right now it is a chore even finding the actual changes.

**Comment by jaseemabid** [comment]
I'm working on this.  I don't have permissions to add WIP label

**Comment by jaseemabid** [comment]
Heads up! most of this is going alright but I hit one ugly issue. The yaml definitions in saas-openshift supports null values for hash key. I assume this means deploying the master or something like that.
 
This makes sense in the context of deployments but makes generating changelogs way harder. 

What do I do for a master..master or null..null case?

**Comment by maxandersen** [comment]
For now - skip it and just add a "latest master deployed" or similar.

These are symbols of problems we need to fix.

**Comment by jaseemabid** [comment]
Thanks for the pointer @sbose78. I'll ignore them for now.

**Comment by jaseemabid** [comment]
Ref https://github.com/openshiftio/saasherder/pull/41

**Comment by maxandersen** [comment]
okey, we now got the basics for generating the log on a given update to saas-openshiftio (great job @jaseemabid ) 

Now how do we get the generation triggered on build and made available somewhere - @vpavlin any good place in the pipeline jobs we can pont @jaseemabid  to hook in a call to saasheder changelog and then have it published somewhere ?

**Comment by jaseemabid** [comment]
@vpavlin ping

**Comment by joshuawilson** [comment]
I'm moving this out of the sprint since it hasn't been updated 2 months.
